File: Java-Core/src/main/java/prompto/utils/ExpressionList.java
package prompto.utils;

import prompto.expression.IExpression;

@SuppressWarnings("serial")
public class ExpressionList extends ObjectList<IExpression>{

	public ExpressionList() {
	}
	
	public ExpressionList(IExpression item) {
		this.add(item);
	}

	public void toDialect(CodeWriter writer) {
		if(this.size()>0) {
			for(IExpression exp : this) {
				exp.toDialect(writer);
				writer.append(", ");
			}
			writer.trimLast(2);
		}
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/utils/ExpressionList.js
var ObjectList = require("./ObjectList").ObjectList;

function ExpressionList(items, item) {
    ObjectList.call(this, items, item);
    return this;
}

ExpressionList.prototype = Object.create(ObjectList.prototype);
ExpressionList.prototype.constructor = ExpressionList;

ExpressionList.prototype.toDialect = function(writer) {
    if (this.length > 0) {
        for (var i = 0; i < this.length; i++) {
            this[i].toDialect(writer);
            writer.append(", ");
        }
        writer.trimLast(2);
    }
};

exports.ExpressionList = ExpressionList;